molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DecoratorConfigurationMetadata.java

```diff
@@ -0,0 +1,45 @@
+package org.molgenis.data.decorator.meta;
+
+import org.molgenis.data.meta.AttributeType;
+import org.molgenis.data.meta.SystemEntityType;
+import org.springframework.stereotype.Component;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.decorator.meta.DecoratorPackage.PACKAGE_DECORATOR;
+import static org.molgenis.data.meta.model.EntityType.AttributeRole.ROLE_ID;
+import static org.molgenis.data.meta.model.Package.PACKAGE_SEPARATOR;
+
+@Component
+public class DecoratorConfigurationMetadata extends SystemEntityType
+{
+	private static final String SIMPLE_NAME = "DecoratorConfiguration";
+	public static final String DECORATOR_CONFIGURATION = PACKAGE_DECORATOR + PACKAGE_SEPARATOR + SIMPLE_NAME;
+
+	public static final String ID = "id";
+	public static final String ENTITY_TYPE_ID = "entityTypeId";
+	public static final String DYNAMIC_DECORATORS = "dynamicDecorators";
+
+	private final DecoratorPackage decoratorPackage;
+	private final DynamicDecoratorMetadata dynamicDecoratorMetadata;
+
+	DecoratorConfigurationMetadata(DecoratorPackage decoratorPackage, DynamicDecoratorMetadata dynamicDecoratorMetadata)
+	{
+		super(SIMPLE_NAME, PACKAGE_DECORATOR);
+		this.decoratorPackage = requireNonNull(decoratorPackage);
+		this.dynamicDecoratorMetadata = requireNonNull(dynamicDecoratorMetadata);
+	}
+
+	@Override
+	public void init()
+	{
+		setLabel("Decorator Configuration");
+		setPackage(decoratorPackage);
+
+		addAttribute(ID, ROLE_ID).setAuto(true).setVisible(false).setLabel("Identifier");
+		addAttribute(ENTITY_TYPE_ID).setNillable(false).setUnique(true).setLabel("Entity Type Identifier");
```

Comment:
Shouldn't this be a XREF to EntityType to avoid dynamic decorators pointing to entity types that don't exist?

Reply:
We don't want to change EntityType and we don't have cascading deletes yet so we decided to do it like this in the meantime. I've added a story for cascading deletes:

http://www.molgenis.org/ticket/5565#ticket

Reply:
Having a dangling decorator won't break anything as it will never be added or looked up.

Reply:
But we don't want dangling stuff right? If we can't use an XREF then we should do the cleanup ourselves.

Reply:
I could add an extra system decorator to EntityType to manage this cleanup, or wait for the cascading delete feature to be implemented. Either's fine by me.